src/tlv/TlvNumber.ts

```diff
@@ -0,0 +1,101 @@
+        const value = reader.readPrimitive(typeLength) as T;
+        this.validate(value);
+        return value;
+    }
+
+    override validate(value: T): void {
+        super.validate(value);
+        if (this.min !== undefined && value < this.min) throw new Error(`Invalid value: ${value} is below the minimum, ${this.min}.`);
+        if (this.max !== undefined && value > this.max) throw new Error(`Invalid value: ${value} is above the maximum, ${this.max}.`);
+    }
+
+    /** Restrict value range. */
+    bound({ min, max }: NumericConstraints<T>): TlvNumericSchema<T> {
+        return new TlvNumericSchema(this.type, this.lengthProvider, maxValue(min, this.min) as T, minValue(max, this.max) as T);
+    }
+}
+
+export type NumericConstraints<T extends number | bigint = number | bigint> = {
+    min?: T,
+    max?: T,
+};
+
+export class TlvShortNumberSchema extends TlvNumericSchema<number> {
+    constructor(
+        type: TlvType.UnsignedInt | TlvType.SignedInt | TlvType.Float,
+        lengthProvider: (value: number) => TlvLength,
+        min?: number,
+        max?: number,
+    ) {
+        super(type, lengthProvider, min, max);
+    }
+
+    override decodeTlvInternalValue(reader: TlvReader, typeLength: TlvTypeLength) {
+        const value = super.decodeTlvInternalValue(reader, typeLength);
+        return typeof value === "bigint" ? Number(value) : value;
+    }
+
+    override bound({ min, max }: NumericConstraints<number>): TlvNumericSchema<number> {
+        return new TlvShortNumberSchema(
+            this.type,
+            this.lengthProvider,
+            maxValue(min, this.min),
+            minValue(max, this.max),
+        );
+    }
+}
+
+export const TlvLongNumberSchema = TlvNumericSchema<number | bigint>;
+
+
+/** Unsigned integer TLV schema. */
+export const TlvFloat = new TlvShortNumberSchema(TlvType.Float, _value => TlvLength.FourBytes, FLOAT32_MIN, FLOAT32_MAX);
+export const TlvDouble = new TlvShortNumberSchema(TlvType.Float, _value => TlvLength.EightBytes);
+export const TlvInt8 = new TlvShortNumberSchema(TlvType.SignedInt, value => TlvCodec.getIntTlvLength(value), INT8_MIN, INT8_MAX);
+export const TlvInt16 = new TlvShortNumberSchema(TlvType.SignedInt, value => TlvCodec.getIntTlvLength(value), INT16_MIN, INT16_MAX);
+export const TlvInt32 = new TlvShortNumberSchema(TlvType.SignedInt, value => TlvCodec.getIntTlvLength(value), INT32_MIN, INT32_MAX);
+export const TlvInt64 = new TlvLongNumberSchema(TlvType.SignedInt, value => TlvCodec.getIntTlvLength(value), INT64_MIN, INT64_MAX);
+export const TlvUInt8 = new TlvShortNumberSchema(TlvType.UnsignedInt, value => TlvCodec.getUIntTlvLength(value), 0, UINT8_MAX);
+export const TlvUInt16 = new TlvShortNumberSchema(TlvType.UnsignedInt, value => TlvCodec.getUIntTlvLength(value), 0, UINT16_MAX);
+export const TlvUInt32 = new TlvShortNumberSchema(TlvType.UnsignedInt, value => TlvCodec.getUIntTlvLength(value), 0, UINT32_MAX);
```

Comment:
`TlvShortNumberSchema` seems like an odd name for a 32-bit type.
I'd suggest just calling the 32-bit or less one `TlvNumberSchema`.

Reply:
Done